src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +704,287 @@ impl<
         Ok(())
     }
 
+    /// This function reads rejection-related information from the exit-at-rc clarity contract.
+    /// It returns true if the rejection succeeded, and false if it failed.
+    pub fn read_rejection_state(
+        &mut self,
+        rc_cycle_of_rejection: u64,
+        proposed_exit_rc: u64,
+        consensus_hash: &ConsensusHash,
+        block_hash: &BlockHeaderHash,
+        exit_contract_id: &QualifiedContractIdentifier,
+    ) -> Result<bool, Error> {
+        let stacks_block_id = StacksBlockHeader::make_index_block_hash(consensus_hash, block_hash);
+        let reward_cycle_length = self.burnchain.pox_constants.reward_cycle_length;
+        let rejection_percent_threshold = self
+            .burnchain
+            .exit_contract_constants
+            .rejection_confirmation_percent;
+        let is_mainnet = self.burnchain.is_mainnet();
+        self.chain_state_db
+            .with_read_only_clarity_tx(&self.sortition_db.index_conn(), &stacks_block_id, |conn| {
+                conn.with_clarity_db_readonly(|db| {
+                    // from map rc-proposal-rejections, use key pair (proposed_rc, curr_rc) to get the # of rejections
+                    let entry = db
+                        .fetch_entry_unknown_descriptor(
+                            exit_contract_id,
+                            "rc-proposal-rejections",
+                            &Value::from(
+                                TupleData::from_data(vec![
+                                    ("proposed-rc".into(), Value::UInt(proposed_exit_rc as u128)),
+                                    ("curr-rc".into(), Value::UInt(rc_cycle_of_rejection as u128)),
+                                ])
+                                .expect("BUG: failed to construct simple tuple"),
+                            ),
+                        )
+                        .expect("BUG: Failed querying rc-proposal-rejections")
```

Comment:
Hmm. What happens if this is evaluated before the contract is launched? Does it panic?